Data Structures/matrix_sum_query.cpp

```diff
@@ -0,0 +1,68 @@
+// Name : Shivam Singhal
+// Email ID : shivamsinghal0610@gmail.com
+// Algorithm to calculate the sum of elements contained in a sub-matrix
+// formed by given 2 co-ordinates in a query taking first point as upper left
+// corner and the second point as the lower right corner
+
+#include <iostream>
+using namespace std;
+
+int main()
+{
+	int m, n, q;
+	cin >> m >> n;
+	cout << "Dimensions of matrix are : " << m <<", "<<n<<endl;
+	int a[m][n];
+	for(int i=0; i<m; i++){
+		for(int j=0; j<n; j++)
+			cin >> a[i][j];
+	}
+	cout << "Matrix elements are :" << endl;
+	for(int i=0; i<m; i++){
+		for(int j=0; j<n; j++)
+			cout << a[i][j] << " ";
+		cout << endl;
+	}
+	for(int i=1; i<m; i++){
+		for(int j=0; j<n; j++){
+			a[i][j] += a[i-1][j];
+		}
+	}
+	for(int i=1; i<m; i++){
+		for(int j=1; j<n; j++){
+			a[i][j] += a[i][j-1];
+		}
+	}
+	for(int i=1 ; i<n ; i++)
+		a[0][i] += a[0][i-1];
+	cout<<"Sum matrix:\n";
+	for(int i=0;i<m;i++){
+		for(int j=0;j<n;j++)
+			cout<<a[i][j]<<" ";
+		cout<<endl;
+	}
+	cin >> q;
+	cout << "Number of queries : " << q << endl;
+	for(int i=1; i<=q; i++){
+		int ai, aj, bi, bj;
```

Comment:
Provide a prompt here also, so that appropriate values are entered.

Reply:
Prompt for the co-ordinates, right ?

Reply:
Yes, mention the user that you want to top left and bottom right coordinates. Also, mention whether you want them to be 0-indexed or 1-indexed.